src/ray/gcs/gcs_server/object_locator.cc

```diff
@@ -0,0 +1,225 @@
+#include "ray/gcs/gcs_server/object_locator.h"
+
+namespace ray {
+
+namespace gcs {
+
+ObjectLocationInfo::ObjectLocationInfo(const ObjectID &object_id)
+    : object_id_(object_id) {}
+
+ObjectLocationInfo::~ObjectLocationInfo() {}
+
+void ObjectLocationInfo::AddLocation(const ClientID &node_id) {
+  locations_.emplace(node_id);
+  RAY_LOG(DEBUG) << "Add object " << object_id_ << " location " << node_id;
+}
+
+size_t ObjectLocationInfo::RemoveLocation(const ClientID &node_id) {
+  auto it = locations_.find(node_id);
```

Comment:
How about locations_.erase directly?

Reply:
OK

Reply:
Algorithm complexity is the same, both methods are fine.